docs/current/faq.md

```diff
@@ -147,6 +147,42 @@ Follow these steps:
     - Python: [https://github.com/platformdirs/platformdirs](https://github.com/platformdirs/platformdirs)
   :::
 
+### Why we do not support rootless engine?
+
+Rootless aims to run a container without `--privileged` flag, meaning that the container has not root capabilities to all devices on the host system.
+Currently, the Dagger engine cannot be run as a rootless container, some complex features such as network, multi-platform and volumes are not supported when running rootless.
+
+We know that it can be a limitation if you are using a managed container environment or have security constraints.
+However, running an unprivileged dagger engine strongly limit capabilities and performances of dagger.
+
+#### Platform constraint
+
+Dagger engine relies on a buildkit engine, it order to run it rootless, we need to use [Rootlesskit](https://github.com/rootless-containers/rootlesskit).
+It's an experimental tool that aims to replace root with a fakeroot using user namespaces, however depending on the host operating system, some feature might not be fully supported.
+
+Meaning that dagger may have unsupported feature on some platform and kernel specifications. Since dagger aims to be a CI/CD that runs consistently everywhere, dealing with specific platform capabilities is opposite to our objective.
+No matter the distribution: Linux, Mac or Windows, Dagger need to work the same way.
+
+#### Volume constraint
+
+Dagger engine relies on overlayfs snapshotter to mounts for efficient operation, running as rootless require to use [fuse-overlayfs](https://github.com/containers/fuse-overlayfs) which is a rootless implementation of overlayfs.
+
+Choosing the right configuration depends on the kernel capabilities, its own configuration, fuse availability and the upstream kernel patches specific to the host distribution.
+Meaning that if it's not possible to use the default snapshotter, you will experience noticeable slowdown, either with fuse-overlayfs or the default snapshotter.
+
+Dagger aims to accelerate your CI/CD pipeline, reducing performances also reduces the value Dagger adds as well as leading to unstable behaviour.
+
+#### Network constraint
+
+Using a rootless engine will especially constraint the custom network management of Dagger.
+
+It is possible to use [slirp](https://github.com/rootless-containers/slirp4netns) as a workaround, but it will highly decrease network performances in addition to be extremely complex to setup.
+Rootlesskit wrote a [table of comparaison]((https://github.com/rootless-containers/rootlesskit/blob/master/docs/network.md#network-drivers)), we can observe that slirp is at least five time slower than root network, meaning that dagger will be way slower if ran as rootless.
+
+#### Conclusion
+
+It might be possible in the future once these constraint vanish but for now it's highly recommended to run dagger engine with privileges if you want to unlock its full potentiel.
```

Comment:
I think this should be stronger - we don't just recommend running the dagger engine with privileges, it's not really possible to do so without.

Reply:
I said highly recommended in case some people really want to, it might be possible but we will not support that decision, I couldn't find a way when I tried but maybe someone with more experience with rootless tech can, I do not know.

I can write it is not possible if you think it's totally impossible

Reply:
I doubt it's totally impossible - but it will require substantial code changes in the engine IMO.

I just want to make sure users don't go away and spend time trying to get it working, if they think it can be done with just a few surface level changes, since that's likely to make people unhappy :cry:

Reply:
Alright! 🚀  Fixed in [a901bc6](https://github.com/dagger/dagger/pull/5809/commits/a901bc6a0cd7a319f0dfed3df420ac112731917e)